src/sdk/namespace_fs.js

```diff
@@ -509,12 +514,13 @@ class NamespaceFS {
     async read_object_md(params, object_sdk) {
         const fs_context = this.prepare_fs_context(object_sdk);
         try {
-            const file_path = this._get_file_path(params);
+            const file_path = await this._get_version_path(params, fs_context);
             await this._check_path_in_bucket_boundaries(fs_context, file_path);
             await this._load_bucket(params, fs_context);
             const stat = await nb_native().fs.stat(fs_context, file_path);
             if (isDirectory(stat)) throw Object.assign(new Error('NoSuchKey'), { code: 'ENOENT' });
-            return this._get_object_info(params.bucket, params.key, stat);
+            this._throw_if_delete_marker(stat);
+            return this._get_object_info(params.bucket, params.key, stat, params.version_id || 'null');
```

Comment:
why do we need this 'null'? 
Also, when bucket versioning is disabled we shouldn't return version_id

Reply:
No it is the past behavior. `_get_object_info()` used to return `version_id` as `null`

Reply:
you are right we used to always return '1'? so the sdk/cli removes it when getting an answer?

Reply:
let me check this